ts_src/payments/taprootutils.ts

```diff
@@ -46,94 +49,63 @@ export interface HashTree {
  *  - one taproot leaf and a list of elements
  */
 export function toHashTree(scriptTree: Taptree): HashTree {
-  if (scriptTree.length === 1) {
-    const script = scriptTree[0];
-    if (Array.isArray(script)) {
-      return toHashTree(script);
-    }
-    script.version = script.version || LEAF_VERSION_TAPSCRIPT;
-    if ((script.version & 1) !== 0)
-      throw new TypeError('Invalid script version');
+  if (isTapleaf(scriptTree)) return { hash: tapleafHash(scriptTree) };
 
-    return {
-      hash: tapLeafHash(script.output, script.version),
-    };
-  }
+  const hashes = [toHashTree(scriptTree[0]), toHashTree(scriptTree[1])];
+  hashes.sort((a, b) => a.hash.compare(b.hash));
+  const [left, right] = hashes;
 
-  let left = toHashTree([scriptTree[0]]);
-  let right = toHashTree([scriptTree[1]]);
-
-  if (left.hash.compare(right.hash) === 1) [left, right] = [right, left];
   return {
     hash: tapBranchHash(left.hash, right.hash),
     left,
     right,
   };
 }
-/**
- * Check if the tree is a binary tree with leafs of type Tapleaf
- */
-export function isTapTree(scriptTree: Taptree): boolean {
-  if (scriptTree.length > 2) return false;
-  if (scriptTree.length === 1) {
-    const script = scriptTree[0];
-    if (Array.isArray(script)) {
-      return isTapTree(script);
-    }
-    if (!script.output) return false;
-    script.version = script.version || LEAF_VERSION_TAPSCRIPT;
-    if ((script.version & 1) !== 0) return false;
-
-    return true;
-  }
-
-  if (!isTapTree([scriptTree[0]])) return false;
-  if (!isTapTree([scriptTree[1]])) return false;
-
-  return true;
-}
 
 /**
  * Given a MAST tree, it finds the path of a particular hash.
  * @param node - the root of the tree
  * @param hash - the hash to search for
- * @returns - and array of hashes representing the path, or an empty array if no pat is found
+ * @returns - and array of hashes representing the path, undefined if no path is found
```

Comment:
We should have a validation inside `p2tr` (and a unit test) if both the `scriptTree` and the `redeem` are provided, but the `redeem` script is not part of the `scriptTree`.

Test Data:
```
    {
      "description": "Redeem script not in tree",
      "exception": "Redeem script not in tree",
      "options": {},
      "arguments": {
        "internalPubkey": "9fa5ffb68821cf559001caa0577eeea4978b29416def328a707b15e91701a2f7",
        "scriptTree": {
          "output": "83d8ee77a0f3a32a5cea96fd1624d623b836c1e5d1ac2dcde46814b619320c18 OP_CHECKSIG"
        },
        "redeem": {
          "output": "83d8ee77a0f3a32a5cea96fd1624d623b836c1e5d1ac2dcde46814b619320c19 OP_CHECKSIG"
        }
      }
    },
```